app/jobs/send_notification_job.rb

```diff
@@ -2,11 +2,32 @@
 
 # Sends a simple one-step notification.
 class SendNotificationJob < ApplicationJob
+  def self.disable_endpoint(endpoint)
+    endpoint.active = false
+    endpoint.save!
+  end
+
+  discard_on(Notifications::FatalError) do |job, error|
+    endpoint = job.endpoint
+    disable_endpoint(endpoint)
```

Comment:
Food for thought: if we disable an endpoint, none of scheduled deliveries after the current one matters anymore. We could leave things as they are and pop them one by one, or maybe we could try to wipe the entire queue.

Reply:
The previous comment was under the impression we use a queue-per-endpoint approach, which was in our early designs. Currently we use one common queue for all the endpoints and for that reason we can't just blow it away. Marking as resolved